js/search/asyncTask.js

```diff
@@ -26,6 +25,20 @@ const AsyncTask = Lang.Class({
     GTypeName: 'EknAsyncTask',
     Extends: GObject.Object,
 
+    Properties: {
```

Comment:
Since the other stuff (`source, cancellable, callback`) is all passed directly to the constructor instead of through a properties dict, I think it would be best to do the same with priority. You could provide a default value for the argument. In any case it won't work as-is, because the `this.parent()` call has to get a properties dict for this property to end up set.

That way you wouldn't have to specify a range for the priority either, which is more in line with what GLib does — you can set any integer as the priority.

I would also make the default value `GLib.PRIORITY_DEFAULT`, I don't think it would be good for all tasks to always run on high priority.